src/summary/history/allergy-card-level-three.component.tsx

```diff
@@ -0,0 +1,225 @@
+  const [patientAllergenReaction, setPatientAllergenReaction] = React.useState(
+    []
+  );
+  const [
+    patientAllergenReactionSeverity,
+    setPatientAllergenReactionSeverity
+  ] = React.useState(null);
+  const [dateOfOnSet, setDateOfOnSet] = React.useState(null);
+  const [comment, setComment] = React.useState(null);
+
+  const handleCategoryReactionChange = event => {
+    setAllergen(null);
+    setCategoryReaction(event.target.value);
+  };
+
+  const handleSubmit = event => {
+    event.preventDefault();
+    alert("SUbmitting the form");
+  };
+
+  React.useEffect(() => {
+    if (categoryReaction) {
+      getAllergyAllergenByConceptUuid(categoryReaction).subscribe(
+        data => setAllergen(data),
+        createErrorHandler
+      );
+      getAllergyReaction().subscribe(
+        data => setAllergyReaction(data),
+        createErrorHandler
+      );
+    }
+  }, [categoryReaction]);
+
+  return (
+    <SummaryCard
+      name="Add Allergy"
+      link=""
+      match={props.match}
+      styles={{
+        width: "100%",
+        background: "var(--omrs-color-bg-medium-contrast)",
+        marginBottom: "2rem",
+        position: "relative",
+        minHeight: "85vh"
+      }}
+    >
+      <form onSubmit={handleSubmit}>
+        <h4 className={`${style.allergyHeader} omrs-bold`}>
+          Category of reaction
+        </h4>
+        <div className={`${style.container}`}>
+          <div className="omrs-radio-button">
+            <label>
+              <input
+                type="radio"
+                name="category"
+                value={DRUGALLERGENCONCEPT}
+                onChange={handleCategoryReactionChange}
+              />
+              <span>Drug</span>
+            </label>
+          </div>
+          <div className="omrs-radio-button">
+            <label>
+              <input
+                type="radio"
+                name="category"
+                value={FOODALLERGENCONCEPT}
+                onChange={handleCategoryReactionChange}
+              />
+              <span>Food</span>
+            </label>
+          </div>
+          <div className="omrs-radio-button">
+            <label>
+              <input
+                type="radio"
+                name="category"
+                value={ENVIROMENTALALLEGENCONCEPT}
+                onChange={handleCategoryReactionChange}
+              />
+              <span>Enviromental</span>
+            </label>
+          </div>
+          <div className="omrs-radio-button">
+            <label>
+              <input
+                type="radio"
+                name="category"
+                value="noAllergy"
+                onChange={handleCategoryReactionChange}
+              />
+              <span>Patient has no allergies</span>
+            </label>
+          </div>
+        </div>
+        {allergen && (
+          <div>
+            <h4 className={`${style.allergyHeader} omrs-bold`}>
```

Comment:
You have this hard-coded for drug allergen. The case for other allergens is forthcoming?